.github/workflows/main.yml

```diff
@@ -0,0 +1,59 @@
+name: Test, Build, Deploy and Api-Test
+
+on:
+  push:
+    branches:
+      - feature/RestApiForSongs
+defaults:
+  run:
+    working-directory: Songify
+    
+jobs:
+  test:
+    name: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Maven Package
+        run: mvn -B clean package
+      - name: Maven Verify
+        run: mvn -B clean verify -DskipTests
+
+  build:
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+      - uses: actions/checkout@v2
+      - uses: akhileshns/heroku-deploy@v3.12.12 # This is the action
+        with:
+          heroku_api_key: ${{secrets.HEROKU_API_KEY}}
+          heroku_app_name: "kbe-songify" #Must be unique in Heroku
+          heroku_email: "y.samet97@gmail.com"
```

Comment:
I recommend not including private emails in your repository.

Reply:
removed email with Github Secrets even though you can see it in my git history :)

Reply:
A solution for this is to delete PR + branch and rewrite git history